ext/java/org/jruby/ext/stringio/StringIO.java

```diff
@@ -296,14 +304,18 @@ private void strioInit(ThreadContext context, int argc, IRubyObject arg0, IRubyO
                     break;
             }
             Encoding encoding = null;
+            Object vmodeVperm = EncodingUtils.vmodeVperm(null, null);
+            int[] oflags = {0};
+            int[] fmode = {0};
```

Comment:
Can we move them to the bellow `if (!options.isNil()) { ... }` block?

Reply:
Fixed in upcoming push, and vmodeVperm and oflags are now allocated once since they are never read by StringIO.